### PR TITLE
[PAC][ELF][AArch64] Support signed personality function pointer

### PR DESCRIPTION
**Pull request by kovdan01**
If function pointer signing is enabled, sign personality function pointer stored in `.DW.ref.__gxx_personality_v0` section with IA key, 0x7EAD = `ptrauth_string_discriminator("personality")` constant discriminator and address diversity enabled.

### CODE REVIEW

**Comment by kovdan01** [comment]
Would be glad to see everyone's feedback on the changes.

**Comment by kovdan01** [comment]
Would be glad to see everyone's feedback on the changes.

**Comment by kovdan01** [comment]
Would be glad to see everyone's feedback on the changes.

File: llvm/lib/Target/AArch64/AArch64TargetObjectFile.cpp [hunk 1]
```diff
@@ -28,6 +29,21 @@ void AArch64_ELFTargetObjectFile::Initialize(MCContext &Ctx,
   SupportDebugThreadLocalLocation = false;
 }
 
+void AArch64_ELFTargetObjectFile::emitPersonalityValueImpl(
+    MCStreamer &Streamer, const DataLayout &DL, const MCSymbol *Sym,
+    const MachineModuleInfo *MMI) const {
+  if (!MMI->getObjFileInfo<MachineModuleInfoELF>().hasSignedPersonality()) {
```
Comment (MaskRay) [inline llvm/lib/Target/AArch64/AArch64TargetObjectFile.cpp:35]: We should add a subclass of MachineModuleInfoELF similar to `AMDGPUMachineModuleInfo`, then move `HasSignedpersonality` there.  Here you can use a static_cast.

File: llvm/lib/Target/AArch64/MCTargetDesc/AArch64TargetStreamer.cpp [hunk 2]
```diff
@@ -35,6 +35,16 @@ AArch64TargetStreamer::AArch64TargetStreamer(MCStreamer &S)
 
 AArch64TargetStreamer::~AArch64TargetStreamer() = default;
 
+void AArch64TargetStreamer::emitAuthValue(const MCExpr *Expr,
+                                          uint16_t Discriminator,
+                                          AArch64PACKey::ID Key,
+                                          bool HasAddressDiversity,
+                                          MCContext &Ctx) {
+  Streamer.emitValueImpl(AArch64AuthMCExpr::create(Expr, Discriminator, Key,
```
Comment (MaskRay) [inline llvm/lib/Target/AArch64/MCTargetDesc/AArch64TargetStreamer.cpp:43]: Use Streamer.getContext() and remove `Ctx` parameter

**Review by MaskRay — APPROVED** [review]

File: llvm/lib/Target/AArch64/AArch64TargetObjectFile.cpp (hunk 1 above)
Comment (kovdan01) [reply llvm/lib/Target/AArch64/AArch64TargetObjectFile.cpp]: > We should add a subclass of MachineModuleInfoELF similar to `AMDGPUMachineModuleInfo`, then move `HasSignedpersonality` there. Here you can use a static_cast.

Thanks for suggestion, please let me know if b4ca8b141b3df8737f0f0d6f64f083fefb403735 resolves your comment

**Comment by kovdan01** [comment]
@MaskRay Please let me know if latest changes fix your previous comments and if this could be merged.

**Pull request merged**